### PR TITLE
Controller for Orders, Users, Authentication and ShoppingCart was added

### PR DESCRIPTION
**Pull request by Mandybur800**

### CODE REVIEW

File: src/main/java/ua/com/conductor/controllers/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @Autowired
+    public AuthenticationController(AuthenticationService authenticationService,
+                                    UserMapper mapper) {
+        this.authenticationService = authenticationService;
+        this.mapper = mapper;
+    }
+
+    @PostMapping("/register")
+    public void registration(@RequestBody UserRequestDto dto) {
+        User user = mapper.toEntity(dto);
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/controllers/AuthenticationController.java]: Why do you need this?

File: src/main/java/ua/com/conductor/controllers/AuthenticationController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        authenticationService.register(dto.getEmail(), dto.getEmail());
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/controllers/AuthenticationController.java]:
```suggestion
        authenticationService.register(dto.getEmail(), dto.getPassword());
```

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/ua/com/conductor/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import ua.com.conductor.model.User;
+import ua.com.conductor.model.dto.UserResponseDto;
+import ua.com.conductor.service.UserService;
+import ua.com.conductor.service.dtomappers.UserMapper;
+
+@RestController
+@RequestMapping("users")
```
Comment (yurwing) [inline src/main/java/ua/com/conductor/controllers/UserController.java]:
```suggestion
@RequestMapping("/users")
```

=)

Comment (Mandybur800) [reply src/main/java/ua/com/conductor/controllers/UserController.java]: Yep, but this work to)

File: src/main/java/ua/com/conductor/controllers/AuthenticationController.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import ua.com.conductor.model.dto.UserRequestDto;
+import ua.com.conductor.security.AuthenticationService;
+import ua.com.conductor.service.dtomappers.UserMapper;
+
+@RestController
+@RequestMapping
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final UserMapper mapper;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/conductor/controllers/AuthenticationController.java]: do you use this field?

File: src/main/java/ua/com/conductor/controllers/AuthenticationController.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    @Autowired
+    public AuthenticationController(AuthenticationService authenticationService,
+                                    UserMapper mapper) {
+        this.authenticationService = authenticationService;
+        this.mapper = mapper;
+    }
+
+    @PostMapping("/register")
+    public void registration(@RequestBody UserRequestDto dto) {
+        authenticationService.register(dto.getEmail(), dto.getEmail());
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/conductor/controllers/AuthenticationController.java]: I agree with Max on his comment, can we fix this?

File: src/main/java/ua/com/conductor/model/dto/OrderResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
+package ua.com.conductor.model.dto;
+
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/conductor/model/dto/OrderResponseDto.java]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/ua/com/conductor/model/dto/ShoppingCartResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package ua.com.conductor.model.dto;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long userId;
+    private List<Long> ticketsId;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/conductor/model/dto/ShoppingCartResponseDto.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/ua/com/conductor/controllers/AuthenticationController.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import ua.com.conductor.model.dto.UserRequestDto;
+import ua.com.conductor.security.AuthenticationService;
+import ua.com.conductor.service.dtomappers.UserMapper;
+
+@RestController
+@RequestMapping
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/conductor/controllers/AuthenticationController.java]: what is the purpose of this annotation here?

File: src/main/java/ua/com/conductor/controllers/AuthenticationController.java (hunk 5 above)
Comment (Mandybur800) [reply src/main/java/ua/com/conductor/controllers/AuthenticationController.java]: I thought wat fix it

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**